src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/gather/gather_kernel_ref.cpp

```diff
@@ -46,11 +46,24 @@ ParamsKey GatherKernelRef::GetSupportedKey() const {
     k.EnableOutputDataType(Datatype::INT8);
     k.EnableOutputDataType(Datatype::UINT8);
     k.EnableInputLayout(DataLayout::bfyx);
-    k.EnableOutputLayout(DataLayout::bfyx);
     k.EnableInputLayout(DataLayout::bfzyx);
-    k.EnableOutputLayout(DataLayout::bfzyx);
     k.EnableInputLayout(DataLayout::bfwzyx);
+    k.EnableInputLayout(DataLayout::b_fs_yx_fsv4);
+    k.EnableInputLayout(DataLayout::b_fs_yx_fsv16);
+    k.EnableInputLayout(DataLayout::b_fs_yx_fsv32);
+    k.EnableInputLayout(DataLayout::b_fs_zyx_fsv16);
+    k.EnableInputLayout(DataLayout::b_fs_zyx_fsv32);
+    k.EnableInputLayout(DataLayout::bs_fs_yx_bsv16_fsv16);
+    k.EnableOutputLayout(DataLayout::bfyx);
+    k.EnableOutputLayout(DataLayout::bfzyx);
     k.EnableOutputLayout(DataLayout::bfwzyx);
+    k.EnableOutputLayout(DataLayout::b_fs_yx_fsv4);
+    k.EnableOutputLayout(DataLayout::b_fs_yx_fsv16);
```

Comment:
Enabled all format.

Reply:
I meant that you can use EnableAllInputLayout or EnableAllOutputLayout instead of adding one by one, if you can ensure that all layouts are supported by this kernel actually.

Reply:
Fixed

Reply:
This PR guarantees only blocked formats(Includes fs_b_yx_fsv32) works. Planar formats which is not the bfyx order like yxfb won't work correctly in this PR(Note: originally not supported). I'll revert enabling all to enabling each format.

Reply:
Reverted

Reply:
Test fails(segmentation fault) in byxf. Previous kernel is written for only default planar formats and I worked only blocked formats, so non-default planar format currently not works.

Reply:
Could you check why the test is failing?

Reply:
Updated to support non-planar format [447c417](https://github.com/openvinotoolkit/openvino/pull/11486/commits/447c4172f319ccd211f405c869b66e4b83a039f7)